src/System.IO.Ports/src/System/IO/Ports/SerialPort.Win32.cs

```diff
@@ -3,36 +3,125 @@
+            // QueryDosDevice involves finding any ports that map to \Device\Serialx (call with null to get all, then iterate to get the actual device name) [Typically maps to COMx value]
+            // https://docs.microsoft.com/en-gb/previous-versions/ff546502%28v%3dvs.85%29
             //
-            // QueryDosDevice involves finding any ports that map to \Device\Serialx (call with null to get all, then iterate to get the actual device name)
+            // GetPortNames will return serial port device names for both registry registered SerialComm devices as well as DosDevice names from Interop QueryDosDevice()
+            // USB Serial Ports on Win10IoT are not initalised in registry key HKLM\HARDWARE\DEVICEMAP\SERIALCOMM correctly, placing garbage chars in this keys value
+            // So to handle this, we are returning a list of port names from both locations.
+            // ToDo: If duplicates wanted to be removed, enumerate the rgistery and try and match & remove duplicate 
+            //   HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Enum\USB is one location, but matching criteria not clear!
+            //   HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\COM Name Arbiter\Devices (empty on Win10IoT)
+
+            HashSet<string> resultPortNames = new HashSet<string>();
+
+            // If running on Windows IoT, search and add the Serial Devices from QueryDosDevice
+            // If this is allowed to happen on Windows devices where port detection happens normally, duplicate Serial Port names are added.
+            // This causes issues with System.IO.Ports.Tests
+            // Port detection broken on Windows IoT (Does not initialise registry with COM port names), so use QueryDosDevice            
+            if (RuntimeInformation.OSArchitecture == Architecture.Arm || RuntimeInformation.OSArchitecture == Architecture.Arm64)
+            {
+                // Query Interop QueryDosDevice()
+                //            
+                // Is GUID class ID for COM ports https://docs.microsoft.com/en-us/windows-hardware/drivers/install/guid-devinterface-comport
+                // Typical DosDevice serial port name 'USB#VID_1B4F&PID_214F&MI_00#6&381731fd&0&0000#{86e0d1e0-8089-11d0-9ce4-08003e301f73}'
+                //
+                foreach (string dosName in QueryDosDeviceComPorts(GUID_DEVINTERFACE_COMPORT))
+                {
+                    resultPortNames.Add(dosName.ToLower());
+                }
+            }
+            else
+            {
+                // Query registry register COM[x] ports
+                //
+                using (RegistryKey serialKey = Registry.LocalMachine.OpenSubKey(@"HARDWARE\DEVICEMAP\SERIALCOMM"))
+                {
+                    if (serialKey != null)
+                        foreach (string valueName in serialKey.GetValueNames())
+                        {
+                            string portName = ((string)serialKey.GetValue(valueName)).ToUpper();
```

Comment:
please use comparison with ignore case and use invariant culture instead of creating new copy of string

Reply:
Am fixing these as i go, but just dawned on me is Culture an issue in regards to Device Names?
As there are a lot of string comparisons done in the tests that are not Culture aware.
Also if user was to change language of PC, would this not mean registry values would have to be re-written?

Reply:
I don't believe these are ever localized.